app/model/questionnaire.py

```diff
@@ -5,8 +9,24 @@ def __init__(self):
         self.survey_id = None
         self.description = None
         self.groups = []
+        self.item_register = {}
+        self.validation = None
 
     def add_group(self, group):
         if group not in self.groups:
             self.groups.append(group)
-            group.questionnaire = self
+            group.container = self
+            self.register_item(group.id, group)
+
+    def register_item(self, item_id, item):
+
+        if item_id in self.item_register.keys():
+            raise QuestionnaireException('Duplicate id found - ' + item_id)
+
+        self.item_register[item_id] = item
+        item.questionnaire = self
```

Comment:
I actually wonder if this method should be on a different class, maybe an 'ItemRegister' class